web/modules/custom/rdf_entity/src/Entity/Query/Sparql/Query.php

```diff
@@ -195,7 +195,7 @@ public function condition($property, $value = NULL, $operator = '=', $langcode =
             $label_list = "(<" . implode(">, <", array_unique(array_keys($mapping[$this->entityTypeId]))) . ">)";
             $this->condition->condition('?entity', '?label_type', '?label');
             $this->filter->filter('?label_type IN ' . $label_list);
-            $this->filter->filter('regex(?label, "' . $value . '", "i")');
+            $this->filter->filter('?label IN ("' . $value . '")');
```

Comment:
The actual problem is kind of a really weird problem that I did not have the time to really look into and it is regarding behat tests.
When creating an entity using a table, (like "Given collections: followed by a table of columns of field values) and in this entity there is a reference field, it seems that the reference field handler does not filter the given values. 
As a result, if a blank value is given, it does not consider it as empty, thus removing it, but tries to load from the database using that value.
The above code had the corresponding mysql query 'WHERE label LIKE '%%' so it matched all entities when there should be none. 
Strict checking with the IN method here works fine. This covers behat and similar cases that we might had run onto in the future.